app/common/lib/ledgerUtil.js

```diff
@@ -303,7 +472,18 @@ const getMediaProvider = (url) => {
 
   // Youtube
   if (url.startsWith('https://www.youtube.com/api/stats/watchtime?')) {
-    provider = ledgerMediaProviders.YOUTUBE
+    return ledgerMediaProviders.YOUTUBE
+  }
+
+  // Twitch
+  if (
+    (
+      (firstPartyUrl && firstPartyUrl.startsWith('https://www.twitch.tv')) ||
```

Comment:
this origin check can be spoofed, e.g. by a request to 'https://www.twitch.tv.evil.com'. you probably want to add a trailing slash or do a proper URL parse.

Reply:
done